### PR TITLE
fix issue #74

### PR DESCRIPTION
**Pull request by sandheep45**

### CODE REVIEW

**Comment by prasenjeet-symon** [comment]
Currently, we are not planning to use the controller that is part of the refactoring. Please proceed to remove the 'controller' folder and instead utilize the single file 'api/index.ts' for managing all the guarded API routes related to the client dashboard.

Furthermore, I'd like to emphasize that our API endpoints adhere to a specific convention: [baseUrl]/server/**. The inclusion of the 'server' keyword is crucial for load balancing and rerouting on the proxy server. This design choice is made because all our ML/AI codes reside on the Python server. I hope this clarifies the rationale behind it.

Regarding the 'getPaginatedUserActivities' function, there is already an empty route created in the file described.

Lastly, I recommend avoiding offset-based pagination and instead opt for cursor-based pagination, which involves specifying both the size and cursor (any column) for more efficient results.

File: packages/server/src/api/index.ts [hunk 1]
```diff
@@ -2700,6 +2700,34 @@ router.get('/user/read-later/unread/:size/:cursor', async (req, res) => {
...
 router.get('/user/article-activities/:size/:cursor', async (req, res) => {
+    const { size, cursor } = req.params;
+
+    if (+cursor <= 0 || +size <= 1) return res.status(400).json({ error: 'cursor cannot be less than 0 and/or size cannot be less than 1' });
+
+    const prisma = PrismaClientSingleton.prisma;
+
+    try {
+        const userActivities = await prisma.userActivity.findMany({
+            skip: 1,
```
Comment (prasenjeet-symon) [inline packages/server/src/api/index.ts:2711]: We cannot use the skip. Sorry about this

File: packages/server/src/api/index.ts [hunk 2]
```diff
@@ -2700,6 +2700,34 @@ router.get('/user/read-later/unread/:size/:cursor', async (req, res) => {
...
  */
 router.get('/user/article-activities/:size/:cursor', async (req, res) => {
+    const { size, cursor } = req.params;
+
+    if (+cursor <= 0 || +size <= 1) return res.status(400).json({ error: 'cursor cannot be less than 0 and/or size cannot be less than 1' });
+
+    const prisma = PrismaClientSingleton.prisma;
+
+    try {
+        const userActivities = await prisma.userActivity.findMany({
```
Comment (prasenjeet-symon) [inline packages/server/src/api/index.ts:2710]: Take the approach from the user side ( user --> article Activities --> ( sort by id ( desc ) ) --> take size .

File: packages/server/src/api/index.ts (hunk 1 above)
Comment (sandheep45) [reply packages/server/src/api/index.ts:2711]: [Here's](https://www.prisma.io/docs/concepts/components/prisma-client/pagination#do-i-always-have-to-skip-1) something regarding this

Comment (prasenjeet-symon) [reply packages/server/src/api/index.ts:2711]: That is a bug of skip ( we are actually querying the db with sorting and taking the ( size ) elements at a time.

Comment (sandheep45) [reply packages/server/src/api/index.ts:2711]: So do you want to include cursor?

Comment (prasenjeet-symon) [reply packages/server/src/api/index.ts:2711]: Yes!

**Review by prasenjeet-symon — APPROVED** [review]
Looks good. Thanks for your valuable contribution.

**Pull request merged**